src/main/java/org/jboss/set/pull/processor/AbstractProcessor.java

```diff
@@ -0,0 +1,168 @@
+
+    /**
+     * Returh phase for which implementation will work. This is used to pick proper evaluators and actions
+     *
+     * @return
+     */
+    public abstract ProcessorPhase getPhase();
+
+    /**
+     * Processor can fetch PRs based on any criteria it sees fit and by any means. {@link AbstractProcessor#fetchPullRequests}
+     * will call it and perform some filtering.
+     *
+     * @return
+     */
+    protected abstract List<PullRequestReference> fetchPullRequestsRaw();
+
+    /**
+     * Provide basic filtering of existing PRs and matching codebase to one present in jboss streams.
+     */
+    private List<PullRequestReference> fetchPullRequests() {
+        //NOTE1: check if we dont leak PRs this way?
+        //NOTE2: do we even care if we leak them?
+        return fetchPullRequestsRaw().stream().filter(pr -> {
+            if (pr.getComponentDefinition().isFound()
+                    && pr.getComponentDefinition().getStreamComponent().getCodebase().equals(pr.getPullRequest().getCodebase()))
+                return true;
+            else
+                return false;
+        }).collect(Collectors.toList());
+    }
+
+    public void process() throws ProcessorException {
+        try {
+            final List<EvaluatorData> processedPullRequests = new ArrayList<>();
+            final List<PullRequestReference> pullRequests = fetchPullRequests();
+            log(Level.INFO, " processing: " + pullRequests.size() + " PRs");
+            List<Future<EvaluatorData>> results = this.processorConfig.getExecutorService()
+                    .invokeAll(pullRequests.stream().map(e -> new PullRequestEvaluatorTask(e.getPullRequest(),e.getComponentDefinition())).collect(Collectors.toList()));
+
+            for (Future<EvaluatorData> result : results) {
+                try {
+                    processedPullRequests.add(result.get());
+                } catch (Exception ex) {
+                    log(Level.SEVERE, "ouch !", ex);
+                }
+            }
+
+            log(Level.INFO, "executing actions:");
+            List<Action> actions = this.processorConfig.getActions();
+            ActionContext actionContext = new ActionContext(this.processorConfig);
+            for (Action action : actions) {
+                log(Level.INFO, "...." + action.getClass().getName());
+                // This means that every processor will have its own set of actions
+                // ie. report write will be per processor
+                action.execute(actionContext, processedPullRequests);
+            }
+        } catch (InterruptedException ex) {
+            throw new ProcessorException("processor execution failed", ex);
+        }
+    }
+
+    protected void log(final Level level, final String msg) {
+        this.logger.log(level, this.simpleName + " " + msg);
+    }
+
+    protected void log(final Level level, final String msg, final Throwable t) {
+        this.logger.log(level, this.simpleName + " " + msg, t);
```

Comment:
warning: The static field AbstractProcessor.logger should be accessed in a static way.